src/Stratis.Features.FederatedPeg/Wallet/DeterministicCoinSelector.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using NBitcoin;
+
+namespace Stratis.Features.FederatedPeg.Wallet
+{
+    /// <summary>
+    /// Based on the DefaultCoinSelector but with randomness removed, and GroupByScriptKey set to false.
+    /// </summary>
+    public class DeterministicCoinSelector : ICoinSelector
+    {
+        public IEnumerable<ICoin> Select(IEnumerable<ICoin> coins, IMoney target)
+        {
+            IMoney zero = target.Sub(target);
```

Comment:
Is there a reason not to use `Money.Zero` here?

Reply:
Nope I copied the code, let me do that